File: price.py
import requests
import json
import time

import sys
#sys.path.append("..")
from pubsub_client import MsgHub

msghub = MsgHub("pubsub.msghub.io", 443, secure=True)
#msghub = MsgHub("pubsub.msghub.io", 58080)
#msghub = MsgHub("198.199.97.15", 12345)

current = int(round(time.time() * 1000))
msghub.erase('bitcoin', {"timeframe": {"max": current - 10800000}})
count = 0

while True:
    try:
        r = requests.get('http://data.mtgox.com/api/1/BTCUSD/ticker', timeout=5)
    
        _current = int(json.loads(r.text)['return']['now']) / 1000
        _price = json.loads(r.text)['return']['last_local']['display']
    
        if _current > current:
            current = _current
            print str(int(_current)) + ", " + _price
            msghub.publish("bitcoin", json.loads(r.text), True)
            msghub.save("bitcoin", json.loads(r.text))

            count = count + 1

        if count > 60:
            msghub.erase('bitcoin', {"timeframe": {"max": current - 10800 * 1000}})
            count = 0
    
        time.sleep(10)
    except Exception as e:
        print e
        time.sleep(1)
        msghub = MsgHub("pubsub.msghub.io", 443, secure=True)

File: price.js
var httpclient = require('https');
var MsgHub = require('/root/pubsub/pubsub-client.js')

var current = new Date().getTime();
msghub = new MsgHub('https://pubsub.msghub.io');
//msghub = new MsgHub('http://198.199.97.15:12345');
msghub.erase('bitcoin', {"timeframe": {"max": current - 21600 * 1000}});


setInterval( function() {
    var options = {
        host: 'api.bitcoinaverage.com',
        path: '/ticker/global/USD/',
//        rejectUnauthorized: false,
        port: 443
    };

    var req = httpclient.get(options, function(res) {
        res.once('data', function(chunk) {
            console.log(chunk.toString());
            var result;

            try {
                result = JSON.parse(chunk);
            } catch (e) {
                console.log(e);
                return;
            }

            var _current = (new Date(result.timestamp)).getTime();
            var _price = result.last;

            if (_current > current) {
                current = _current;
                console.log(new Date(_current).toString() + ": " + _price);
                msghub.publish("bitcoin", result, true);
                msghub.save("bitcoin", result);
                msghub.erase('bitcoin', {"timeframe": {"max": current - 21600 * 1000}});
            }
        });
    }).on('error', function(e) {
        console.log("Got error: " + e.message);
    });

    global.gc();
}, 30000);

process.on('uncaughtException', function globalErrorCatch(error, p){
    console.error(error);
    console.error(error.stack);
});
